File: pvtapp/src/main/java/com/freshm/pvtapp/service/OperationsService.java
package com.freshm.pvtapp.service;

import java.util.List;

import com.freshm.pvtapp.dto.*;

public interface OperationsService {
    // Receiving
    MaterialReceiptResponse createReceipt(MaterialReceiptRequest request);
    List<MaterialReceiptResponse> getAllReceipts();
    MaterialReceiptResponse getReceipt(Long id);
    void deleteReceipt(Long id);

    // Returns (reject)
    MaterialReturnResponse createReturn(MaterialReturnRequest request);
    List<MaterialReturnResponse> getAllReturns();
    void deleteReturn(Long id);

    // Sorting + packing
    PackingResponse createPacking(PackingRequest request);
    List<PackingResponse> getAllPacking();
    void deletePacking(Long id);
}

File: freshm-frontend/src/services/operationsService.js
import api from "./api";

const unwrap = (res) => res.data?.data ?? res.data ?? [];

// Receiving
export const createReceipt = async (payload) => unwrap(await api.post("/operations/receipts", payload));
export const getReceipts   = async () => unwrap(await api.get("/operations/receipts"));
export const getReceipt    = async (id) => unwrap(await api.get(`/operations/receipts/${id}`));
export const deleteReceipt = async (id) => unwrap(await api.delete(`/operations/receipts/${id}`));

// Returns (reject)
export const createReturn  = async (payload) => unwrap(await api.post("/operations/returns", payload));
export const getReturns    = async () => unwrap(await api.get("/operations/returns"));
export const deleteReturn  = async (id) => unwrap(await api.delete(`/operations/returns/${id}`));

// Sorting + packing
export const createPacking = async (payload) => unwrap(await api.post("/operations/packing", payload));
export const getPacking    = async () => unwrap(await api.get("/operations/packing"));
export const deletePacking = async (id) => unwrap(await api.delete(`/operations/packing/${id}`));
